app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/UpdateRecordRequest.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.request
+
+import play.api.libs.functional.syntax.{toApplicativeOps, toFunctionalBuilderOps}
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.Reads.verifying
+import play.api.libs.json.{JsPath, OWrites, Reads}
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.eis.Assessment
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport.Reads.lengthBetween
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport.isValidCountryCode
+
+import scala.Function.unlift
+
+case class UpdateRecordRequest(
+  eori: String,
+  recordId: String,
+  actorId: String,
+  traderRef: Option[String],
+  comcode: Option[String],
+  goodsDescription: Option[String],
+  countryOfOrigin: Option[String],
+  category: Option[Int],
+  assessments: Option[Seq[Assessment]],
+  supplementaryUnit: Option[Int],
+  measurementUnit: Option[String],
+  comcodeEffectiveFromDate: Option[String],
+  comcodeEffectiveToDate: Option[String]
+)
+
+object UpdateRecordRequest {
+
+  implicit val reads: Reads[UpdateRecordRequest] =
```

Comment:
if you do not do any thing with the field you do not need to write a witer and reader like that, but you can get all this with one line of code:

implicit val format: OFormat[UpdateRecordRequest] = Json.format[UpdateRecordRequest]

Reply:
The validation rules are set using the explicit reader and writer